pkg/cmd/provisioning/aws/delete.go

```diff
@@ -76,7 +78,100 @@ func deleteOIDCBucket(client aws.Client, bucketName, namePrefix string) error {
 			break
 		}
 	}
+	return nil
+}
+
+// deleteCloudFrontOriginAccessIdentity deletes the CloudFront origin access identities if created
+func deleteCloudFrontOriginAccessIdentity(client aws.Client, namePrefix string) error {
+	listCloudFrontOriginAccessIdentitiesOutput, err := client.ListCloudFrontOriginAccessIdentities(&cloudfront.ListCloudFrontOriginAccessIdentitiesInput{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to fetch a list of CloudFront origin access identities")
+	}
+	for _, originAccessIdentity := range listCloudFrontOriginAccessIdentitiesOutput.CloudFrontOriginAccessIdentityList.Items {
+		if *originAccessIdentity.Comment == fmt.Sprintf("%s/%s", ccoctlAWSResourceTagKeyPrefix, namePrefix) {
+			getCloudFrontOriginAccessIdentityOutput, err := client.GetCloudFrontOriginAccessIdentity(&cloudfront.GetCloudFrontOriginAccessIdentityInput{
+				Id: originAccessIdentity.Id,
+			})
+			if err != nil {
+				return errors.Wrapf(err, "failed to get the CloudFront origin access identity with ID %s", *originAccessIdentity.Id)
+			}
+
+			_, err = client.DeleteCloudFrontOriginAccessIdentity(&cloudfront.DeleteCloudFrontOriginAccessIdentityInput{
+				Id:      originAccessIdentity.Id,
+				IfMatch: getCloudFrontOriginAccessIdentityOutput.ETag,
+			})
+			if err != nil {
+				return errors.Wrapf(err, "failed to delete the CloudFront origin access identity with ID %s", *originAccessIdentity.Id)
+			}
+			log.Printf("CloudFront origin access identity with ID  %s deleted", *originAccessIdentity.Id)
+		}
+	}
+	return nil
+}
+
+// deleteCloudFrontDistribution deletes the CloudFront distribution if created
+func deleteCloudFrontDistribution(client aws.Client, namePrefix string) error {
+	ListCloudFrontDistributionsOutput, err := client.ListCloudFrontDistributions(&cloudfront.ListDistributionsInput{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to fetch a list of CloudFront distributions")
+	}
+	for _, distribution := range ListCloudFrontDistributionsOutput.DistributionList.Items {
+		listTagsForCloudFrontResourceOutput, err := client.ListTagsForCloudFrontResource(&cloudfront.ListTagsForResourceInput{
+			Resource: distribution.ARN,
+		})
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch tags for CloudFront distribution with ID %s", *distribution.Id)
+		}
+
+		for _, tag := range listTagsForCloudFrontResourceOutput.Tags.Items {
+			if *tag.Key == fmt.Sprintf("%s/%s", ccoctlAWSResourceTagKeyPrefix, namePrefix) {
+				getCloudFrontDistributionOutput, err := client.GetCloudFrontDistribution(&cloudfront.GetDistributionInput{
+					Id: distribution.Id,
+				})
+				if err != nil {
+					return errors.Wrapf(err, "failed to get CloudFront Distribution with ID %v", distribution.Id)
```

Comment:
Nit for consistency with logging the ID.

```suggestion
					return errors.Wrapf(err, "failed to get CloudFront Distribution with ID %s", *distribution.Id)
```